src/elements/content-preview/ContentPreview.js

```diff
@@ -354,6 +372,24 @@ class ContentPreview extends React.PureComponent<Props, State> {
         } else if (hasTokenChanged) {
             this.updatePreviewToken();
         }
+
+        if (!prevFile && !!file && !selectedVersion && activeAnnotationId) {
+            // Replace any deep linked annotation fileVersionId with the file's current version because the
+            // current functionality is to load the current version of a deep linked annotation on a previous version
+            // on a hard load.
+            // NOTE: When ContentPreview can support mounting with an older file version this behavior can
+            // likely go away
+            history.replace(`/activity/annotations/${currentFileVersionId}/${activeAnnotationId}`);
+        } else if (prevSelectedVersion !== selectedVersion && activeAnnotationId) {
+            // If the selected version has changed, update the deep linked annotation's fileVersionId
+            history.replace(`/activity/annotations/${previewedFileVersionId}/${activeAnnotationId}`);
+        } else if (activeAnnotationId !== prevActiveAnnotationId) {
+            // If the active annotation has changed, update the deep linked annotationId
+            const path = activeAnnotationId
+                ? `/activity/annotations/${previewedFileVersionId}/${activeAnnotationId}`
+                : '/activity/';
+            history.replace(path);
+        }
```

Comment:
Should this logic reside in a function and then called from `componentDidUpdate`